### ISSUE TITLE
Update org-roam-extract-subtree to insert a link to the newly created node

### ISSUE THREAD

**Issue by Haxxflaxx** [org-roam/org-roam#2365]
### Brief Abstract
It would be useful if org-roam-extract-subtree left behind a link to the newly created node.

### Long Description
I've been looking at adding some advice around the org-roam-extract-subtree function, as I in most cases want to reference the note from where I extracted it from.
I think this would be fairly simple to do, though maybe a bit fragile the way I'm envisioning it (my elisp skills are meager at best). I believe it would be easier and more robust to update the function itself, as well as I could see how other people could benefit from it too.

### Proposed Implementation (if any)
At the end of the function we could return to the starting buffer and insert a roam link at point using the stored node name.
If we don't want to change the default behavior, then maybe take an argument, or look at some org-roam-extract-subtree-insert-link variable, that we can set.

### Please check the following:

- [X] No similar feature requests

**Comment by jmay** [comment]
I've been thinking about the same thing. Perhaps enable this behavior when the function is invoked with a prefix argument? That would leave the current behavior unchanged when no argument is provided.

Should the created link be inserted alone on a line where the previous entry was extracted? Or create a header at the same level as the extracted entry?

**Comment by Haxxflaxx** [comment]
Good point with the heading. From an outlining standpoint you probably would want the heading to remain. Also when using org-transclusion you would want the structure to be the same.

**Comment by jmay** [comment]
Maybe there should be a hook function here. After a node has been extracted, pass that information to the hook which can inject a text link, or a heading, or a `#+transclude:` line.

**Comment by jmay** [comment]
@Haxxflaxx Does this do what you need?

With the PR above, you can inject references with
```
(add-hook 'org-roam-post-extraction-functions #'my/link-extracted-node)

(defun my/link-extracted-node (node)
  "Paste link to NODE at current point."
  (let* (
         (id (org-roam-node-id node))
         (description (org-roam-node-formatted node))
         )
    (org-insert-heading)
    (org-do-demote)
    (insert (org-link-make-string
             (concat "id:" id)
             description))
    ))

(defun my/transclude-node (node)
  "Transclude NODE at current point."
  (let* (
         (id (org-roam-node-id node))
         (description (org-roam-node-formatted node))
         )
    (insert (format "#+transclude: %s :level %d"
                    (org-link-make-string
                     (concat "id:" id)
                     description)
                    (+ 1 (org-outline-level))))
    ))
```

**Comment by Haxxflaxx** [comment]
I've tried running running it, but I'm running into problems. I'm getting warnings about sql uniqueness constraints being violated.
I haven't had much time to look into what's going on. I'll come back when I've done some digging.

**Comment by jmay** [comment]
Uniqueness error probably means that there is somewhere where I neglected a `save-buffer` call. When moving a node from one file to another you have to save the source file (minus the node being transferred) first so that the database update hook will remove the node ID before inserted it into the new file.

Maybe it shouldn't preserve the original node ID. Then we can save all the buffers together at the end. There will be a brief moment where there are two nodes in the database with the same title, but then the original one will be deleted.

This is an excuse to write some automated tests and learn how `buttercup` works.

**Comment by Haxxflaxx** [comment]
Today I'm not getting any problems with database, so I'm not sure what that was about. But after digging a little I notice that the hook I've set don't seem to be called. It seems like the node that is passed is empty, so maybe that's the reason why.

**Comment by jmay** [comment]
Did some more refactoring & added tests, PR is updated.

It shouldn't be possible for the passed node to be empty, because then nothing would have been extracted. But I'm sure I haven't thought of all the possible scenarios. Could you create an example file that demonstrates the issue?

**Comment by Haxxflaxx** [comment]
Looking at your latest commit it doesn't seem like `org-roam-post-extraction-functions` is called from anywhere.

I also noticed that the heading name isn't used for naming the file it is broken out to. It seems to reuse the title of the current node. I was testing from a daily note, so it created a new note for that day and extracted into there.

**Comment by jmay** [comment]
Oops, that's my error, I renamed the hook function but never committed that changed. Added to the PR.

I see your issue. In my test cases every subtree is a node, so it is extracting the entire node when you presumably only want the current org entry. That's an oversight, the function ought to be only extracting the current entry and turning it into a node in the target file. I'll fix that now.

**Comment by jmay** [comment]
@Haxxflaxx Latest changes are on the PR at https://github.com/org-roam/org-roam/pull/2366
